File: 004.clj
; Find the largest palindrome made from the product of two 3-digit numbers.

(defn palindrome? [n]
  (= 
    (reverse (str n)) 
    (seq (str n))))

(apply max (filter palindrome? (for [x (range 100 1000) y (range 100 1000)] (* x y))))

File: 004.hs
-- Find the largest palindrome made from the product of two 3-digit numbers.

module Main where


palindrome :: Integer -> Bool
palindrome p
  | show p == reverse (show p) = True
  | otherwise = False


main = putStrLn (show answer)
answer = maximum $ filter palindrome $ map (\[x,y] -> x * y) $ sequence [[100..999], [100..999]]

-- 906609
